import "google/protobuf/descriptor.proto";

package maid.proto;

message MaidMethodOptions
{
	// if notify set true, rpc method one-way mode active.
	//
	// returns(XX) will be ingored:
	// Service::Message* response will be NULL.
	// Service::Closure* done will do nothing.
	// Service_Stub::CallMethod response == NULL && done == NULL is required.
	// Service_Stub::Message* request success send to destination is unknown
	optional bool notify = 1;

    //
	//optional int64 time_out = 2 [default = -1]; // -1: never time out
}

extend google.protobuf.MethodOptions
{
	optional MaidMethodOptions method_options = 1001;
}

//message MaidServiceOptions
//{
//	optional bool notify = 1;
//}
//
//extend google.protobuf.ServiceOptions
//{
//	optional MaidServiceOptions service_options = 1001;
//}
